### ISSUE TITLE
Allow import.meta with module: es2020

### ISSUE THREAD

**Issue by saschanaz** [microsoft/TypeScript#41249]
**TypeScript Version:**  4.0.3

**Search Terms:** import.meta es2020

**Code**

```ts
export function foo() {
  return import.meta.url;
}
```

**Expected behavior:** It should work

**Actual behavior:** `The 'import.meta' meta-property is only allowed when the '--module' option is 'esnext' or 'system'.ts(1343)
`

**Playground Link:** 

**Related Issues:**

**Comment by DanielRosenwasser** [comment]
Idea: allow the code when the module target is `es2020`, change the error message to

```
The 'import.meta' meta-property is only allowed when the '--module' option is 'es2020', 'esnext', or 'system'.
```

**Comment by DanielRosenwasser** [comment]
Actually, is this ES2020? Or ES2021?

**Comment by saschanaz** [comment]
This is ES2020: https://github.com/tc39/proposals/commit/0c27379ebb525ffebed75584827ce51687af99f8#diff-af1d66eb7dbbf6f66e871d26bcad07076a557256a957f558ca21e60924e2b0b7 (it joined after the original `import()` did in 2019)

**Comment by DanielRosenwasser** [comment]
Oof - I accidentally merged this for 4.1, but the change is extremely small...

**Comment by DanielRosenwasser** [comment]
I'll review with others on the team.

Also, thanks @a-tarasyuk!
